File: 21주차/캐시.py
from collections import deque
def solution(cacheSize, cities):
    answer = 0
    c = deque()
    if cacheSize == 0:
        return len(cities)*5
    for i in cities:
        i = i.lower()
        if i in c:
            answer += 1
            c.remove(i)
            c.append(i)
        else:
            answer += 5
            if len(c) < cacheSize:
                c.append(i)
            else:
                c.popleft()
                c.append(i)
    
    return answer
# 캐시 사이즈가 0 인 경우는 항상 5만큼 걸리는 것에 유의

File: 47주차/캐시.js
function solution(cacheSize, cities) {
  var answer = 0;
  if (cacheSize === 0) return 5 * cities.length;
  let cache = [];
  for (let c of cities) {
    c = c.toLowerCase();
    if (cache.includes(c)) {
      //LRU 기반.
      answer += 1;
      let idx = cache.indexOf(c); // 해당 도시 의 인덱스 반환.
      cache.splice(idx, 1); // 삭제하기.
      cache.push(c); // 다시 맨 뒤로 추가.
    } else {
      answer += 5;
      if (cache.length < cacheSize) cache.push(c);
      else {
        cache.shift();
        cache.push(c);
      }
    }
  }
  return answer;
} // 프로그래머스 카카오 - 캐시.
